### PR TITLE
add/a2p product variations api-core-tests

### PR DESCRIPTION
**Pull request by nigeljamesstevenson**
### All Submissions:

-   [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
-   [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
-   [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change?

### Changes proposed in this Pull Request:
Update playwright api-core-tests for shipping crud operations

Closes 362-gh-woocommerce/woocommerce-quality .

- [ ] This PR is a very minor change/addition and does not require testing instructions (if checked you can ignore/remove the next section).

### How to test the changes in this Pull Request:

1. Pull down the code
2. Navigate to woocommerce/plugins/woocommerce
3. Execute tests with pnpm test:api-pw
4. All tests including the new product variations tests should execute and pass successfully

### Other information:

-   [x] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [x] Have you written new tests for your changes, as applicable?
-   [x] Have you created a changelog file for each project being changed, ie `pnpm --filter=<project> changelog add`?

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

File: plugins/woocommerce/tests/api-core-tests/tests/products/products-crud.test.js [hunk 1]
```diff
@@ -1098,25 +1098,183 @@ test.describe('Products API tests: CRUD', () => {
...
+			const responseJSON = await response.json();
+			expect(response.status()).toEqual(200);
+			expect(responseJSON.id).toEqual(productVariationId);
+			expect(responseJSON.regular_price).toEqual('29.00');
+		});
+
+		test('can retrieve all product variations', async ({
+			request
+		}) => {
+			// call API to retrieve all product tags
```
Comment (zhongruige) [inline plugins/woocommerce/tests/api-core-tests/tests/products/products-crud.test.js]: These comments refer to product tags, but we're working with variations here--so we need to update these accordingly. This line and `1162`, `1177`, and `1187` looks like all need to be updated.

**Review by zhongruige — CHANGES_REQUESTED** [review]
Nice work on this @nigeljamesstevenson!

Couple comments we need to update.

Side note, this file is getting quite long. We should look into what it might take to break this out into smaller files so it's easier to read through what is being tested and update accordingly. Nothing we need to do on this PR, but we'll want to start thinking about it, I think.

File: plugins/woocommerce/tests/api-core-tests/tests/products/products-crud.test.js (hunk 1 above)
Comment (nigeljamesstevenson) [reply plugins/woocommerce/tests/api-core-tests/tests/products/products-crud.test.js]: Fully agree @zhongruige! This file is getting large. A consideration is do we break it out into separate files and do some extra data setup in each of these files.. I'll mull it over and ping you on this.

Also - Thanks for pointing out the comments - I actually noticed a couple of others that I will update also - the shame of a copy-paste 😉

Comment (zhongruige) [reply plugins/woocommerce/tests/api-core-tests/tests/products/products-crud.test.js]: Sounds good @nigeljamesstevenson! Thanks for getting in these updates.

**Review by zhongruige — APPROVED** [review]
Nice work on this @nigeljamesstevenson! These ran and tested great for me locally:

Screen Shot 2022-11-01 at 07 49 32

**Pull request merged**